File: zul/src/org/zkoss/zul/api/Listfoot.java
/* Listfoot.java

{{IS_NOTE
	Purpose:
		
	Description:
		
	History:
		Tue Oct 22 14:45:31     2008, Created by Flyworld
}}IS_NOTE

Copyright (C) 2008 Potix Corporation. All Rights Reserved.

{{IS_RIGHT
	This program is distributed under GPL Version 3.0 in the hope that
	it will be useful, but WITHOUT ANY WARRANTY.
}}IS_RIGHT
 */
package org.zkoss.zul.api;


;
/**
 * A row of {@link Listfooter}.
 * 
 * <p>
 * Like {@link Listhead}, each listbox has at most one {@link Listfoot}.
 * <p>
 * Default {@link #getZclass}: z-listfoot (since 5.0.0)
 * 
 * @author tomyeh
 * @since 3.5.2
 */
public interface Listfoot extends org.zkoss.zul.impl.api.XulElement {
	/**
	 * Returns the list box that it belongs to.
	 * <p>
	 * It is the same as {@link #getParent}.
	 */
	public org.zkoss.zul.api.Listbox getListboxApi();

}

File: zul/src/main/resources/web/js/zul/sel/Listfoot.ts
/* Listfoot.ts

	Purpose:

	Description:

	History:
		Tue Jun  9 18:03:06     2009, Created by jumperchen

Copyright (C) 2009 Potix Corporation. All Rights Reserved.

This program is distributed under LGPL Version 2.0 in the hope that
it will be useful, but WITHOUT ANY WARRANTY.
*/
/**
 * A row of {@link Listfooter}.
 *
 * <p>Like {@link Listhead}, each listbox has at most one {@link Listfoot}.
 * @defaultValue {@link getZclass}: z-listfoot
 */
@zk.WrapClass('zul.sel.Listfoot')
export class Listfoot extends zul.Widget {
	override parent!: zul.sel.Listbox | undefined;
	/**
	 * @returns the list box that it belongs to.
	 */
	getListbox(): zul.sel.Listbox | undefined {
		return this.parent;
	}

	//bug #3014664
	override setVflex(vflex?: boolean | string): this { //vflex ignored for Listfoot
		const vflex0 = false;
		return super.setVflex(vflex0);
	}

	//bug #3014664
	override setHflex(hflex?: boolean | string): this { //hflex ignored for Listfoot
		const hflex0 = false;
		return super.setHflex(hflex0);
	}

	/** @internal */
	override deferRedrawHTML_(out: string[]): void {
		out.push('<tr', this.domAttrs_({domClass: true}), ' class="z-renderdefer"></tr>');
	}
}